emark/message.py

```diff
@@ -129,8 +129,28 @@ def update_url_params(self, url, **params):
 
     def inject_utm_params(self, md, **utm):
         for url in INLINE_LINK_RE.findall(md):
+            try:
+                url_parts = parse.urlparse(url)
+                if url_parts.scheme and url_parts.scheme not in [
+                    "http",
+                    "https",
+                    "localhost",
```

Comment:
This should be a STATIC array and simply include '' (emptry string), to catch missing schemes.